Feature/number guessing refactor

mercurrie opened the pull request:
created art.py
created flowchart for the number guessing game
created main.py to import logo from art.py, and def and call functions needed to create the program to desired parameters

mercurrie committed:
Added random_number_guesser.drawio.png

mercurrie committed:
Update random_number_guesser.drawio.png

mercurrie committed:
completed number_guessing_game

added art.py which contains ASCII art
added main.py which imports logo from art.py, defines and calls functions level_of_difficulty() and play()

michaelhousman commented on main.py:
```diff
@@ -0,0 +1,41 @@
+    
+    while attempts > 0 and guess != answer:
+        print(f'\nYou have {attempts} attempts remaining to guess the number. ')
+        guess = int(input('Pick a number. '))
```
I would abstract this into its own function that gets called by `play()`

michaelhousman replied:
Much better

michaelhousman commented on main.py:
```diff
@@ -0,0 +1,41 @@
+
+# level_of_difficulty() returns number of attempts based on difficulty selection
+def level_of_difficulty(choice):
+    if choice == 'easy':
```
In general, I strongly prefer to see dictionaries / mappings instead of if statements.  Please use a dictionary here instead.

michaelhousman commented on main.py:
```diff
@@ -0,0 +1,41 @@
+from art import logo
+
+# level_of_difficulty() returns number of attempts based on difficulty selection
+def level_of_difficulty(choice):
```
Some comment about type-casting with blackjack functions.  Let's do it here too.

michaelhousman reviewed (changes requested):
Added a few comments in here

mercurrie committed:
refactor code

turned while loop that handles the user's guesses into function user_guesses()

included type cast to functions

changed if statement in level_of_difficulty(() into Dict

michaelhousman commented on main.py:
```diff
@@ -0,0 +1,40 @@
+from art import logo
+
+# level_of_difficulty() returns number of attempts based on difficulty selection
+def level_of_difficulty(choice: str)-> str:
```
Good

michaelhousman reviewed (approved):
Much better.  Btw, what IDE do you use?  I use PyCharm bc you'll see PEP-8 (look it up) suggestions in line with the code, which is super helpful.
